test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/DirectoryInfoAssertions.cs

```diff
@@ -56,6 +56,15 @@ public AndConstraint<DirectoryInfoAssertions> HaveFiles(IEnumerable<string> expe
             return new AndConstraint<DirectoryInfoAssertions>(this);
         }
 
+        public AndConstraint<DirectoryInfoAssertions> HaveFilesMatching(string expectedFilesSearchPattern, SearchOption searchOption)
+        {
+            var matchingFileExists = _dirInfo.EnumerateFiles(expectedFilesSearchPattern, searchOption).Any();
+            Execute.Assertion.ForCondition(matchingFileExists == true)
+                .FailWith("Expected directory {0} to contain files matching {1}, but no matching file exists.",
+                    _dirInfo.FullName, expectedFilesSearchPattern);
+            return new AndConstraint<DirectoryInfoAssertions>(this);
+        }
+
         public AndConstraint<DirectoryInfoAssertions> NotHaveFiles(IEnumerable<string> expectedFiles)
```

Comment:
You're right, all those are super useful. will look into it in the next weeks.
Also, the `TempFileSystem` namespace would probably make an awesome NuGet package on its own..